cluster/gce/gci/credential-provider/README.md

```diff
@@ -0,0 +1,35 @@
+# GCP credential provider for e2e testing
```

Comment:
I haven't had a chance to review the approach (will largely defer to other reviewers), but we should not be adding new things under cluster... if this is test only, can it live under `test/` somewhere, or under legacy-cloud-providers gce?

Reply:
I was mainly following what `cluster/gce/gci/mounter` was doing -- I'll see if I can move this somewhere under `test/`. I tried to avoid legacy-cloud-providers/gce because in theory we may want this specific test around even after we remove cloud providers (because it tests the plugin mechanism without needing to import cloud libraries)

Reply:
I had actually thought we'd already moved `mounter` source out, but realized there was only an issue open to do so: https://github.com/kubernetes/kubernetes/issues/103866